### DISCUSSION TITLE
"Depict error: Substructure match with reference not found" on visually similar molecule

### DISCUSSION THREAD

**Discussion by cbwchuck** [rdkit/rdkit#7856] [Q&A] [answered]
I am trying to align two molecules with `AllChem.GenerateDepictionMatching2DStructure`, however, `ValueError: Depict error: Substructure match with reference not found.` was thrown. I am not sure whether it's expected behaviour.

```python
from rdkit import Chem
from rdkit.Chem import Draw, AllChem
from rdkit.Chem.Draw import IPythonConsole
from rdkit.Chem.Draw import rdDepictor

mol1 = Chem.MolFromSmiles('O=C(Nc1cn2ccnc2c(N2CCC(F)(F)CC2)n1)c1ccc(NS(=O)(=O)CCO)cc1N1CCC2(CC1)CC2')
mol2 = Chem.MolFromSmiles('COc1ccc(NC(=O)c2ccc(NS(=O)(=O)CCO)cc2N2CCC3(CC2)CC3)nc1N1CCC(F)(F)CC1')

AllChem.Compute2DCoords(mol1)
AllChem.GenerateDepictionMatching2DStructure(mol2, mol1, acceptFailure=False)
Draw.MolsToGridImage([mol1, mol2])
```
When `acceptFailure=True`, the two molecules look like align-able.

<img width="424" alt="image" src="https://github.com/user-attachments/assets/99725883-7beb-4e19-b5e8-48f3816d07b5">

**Comment by ptosco** [comment, marked as answer]
`GenerateDepictionMatching2DStructure` expects `mol1` to be a substructure of `mol2`. If that is not the case, you need to either supply a `refPatt` query molecule to specify which common atoms should be mapped, or an explicit `(queryIdx, molIdx)` atom mapping. You may easily obtain this from an MCS calculation:
```
from rdkit import Chem
from rdkit.Chem.rdRascalMCES import FindMCES
from rdkit.Chem import Draw, rdDepictor

mol1 = Chem.MolFromSmiles('O=C(Nc1cn2ccnc2c(N2CCC(F)(F)CC2)n1)c1ccc(NS(=O)(=O)CCO)cc1N1CCC2(CC1)CC2')
mol2 = Chem.MolFromSmiles('COc1ccc(NC(=O)c2ccc(NS(=O)(=O)CCO)cc2N2CCC3(CC2)CC3)nc1N1CCC(F)(F)CC1')

res = FindMCES(mol1, mol2)
assert res
matches = res[0].atomMatches()
smarts = res[0].smartsString

# explicit atom mapping
rdDepictor.Compute2DCoords(mol1)
rdDepictor.GenerateDepictionMatching2DStructure(mol2, mol1, atomMap=matches)
Draw.MolsToGridImage([mol1, mol2])
```
![image](https://github.com/user-attachments/assets/b90e1173-8c9f-440d-80ef-622a2248711c)
```
# SMARTS refPatt
rdDepictor.GenerateDepictionMatching2DStructure(mol2, mol1, refPatt=Chem.MolFromSmarts(smarts));
Draw.MolsToGridImage([mol1, mol2])
```
![image](https://github.com/user-attachments/assets/87d1c662-b797-47d7-aafb-ff58007edccd)

**Reply by cbwchuck** [reply]
Thanks so much @ptosco !
